mslib/mscolab/server.py

```diff
@@ -351,6 +351,38 @@ def get_user():
     return json.dumps({'user': {'id': g.user.id, 'username': g.user.username}})
 
 
+@APP.route('/upload_profile_image', methods=["POST"])
+@verify_user
+def upload_profile_image():
+    user_id = request.form['user_id']
+    file = request.files['image']
+    if not file:
+        return jsonify({'message': 'No file provided or invalid file type'}), 400
+    if not file.mimetype.startswith('image/'):
+        return jsonify({'message': 'Invalid file type'}), 400
+    if file.content_length > mscolab_settings.MAX_UPLOAD_SIZE:
+        return jsonify({'message': 'File too large'}), 413
+
+    success, message = fm.save_user_profile_image(user_id, file)
+    if success:
+        return jsonify({'message': message}), 200
+    else:
+        return jsonify({'message': message}), 400
+
+
+@APP.route('/fetch_profile_image', methods=["GET"])
+@verify_user
+def fetch_profile_image():
+    user_id = request.form['user_id']
+    user = User.query.get(user_id)
+    if user and user.profile_image_path:
+        base_path = mscolab_settings.UPLOAD_FOLDER
+        filename = user.profile_image_path
+        return send_from_directory(base_path, filename)  # todo : Supply os path semantics in this and def uploads()
```

Comment:
the todo is what it is currently. 

A ToDo means there can be done something I ack this but we do it later.

so you better write

# ToDo needs to use fs for the path handling and send_directory needs os.path see discussion on #2103